internal/otelcollector/config/metric/gateway/config_builder.go

```diff
@@ -86,108 +88,127 @@ func makeServiceConfig() config.Service {
 	maps.Copy(envVars, otlpExporterEnvVars)
 
-	otlpExporterID := otlpexporter.ExporterID(pipeline.Spec.Output.Otlp, pipeline.Name)
-	cfg.Exporters[otlpExporterID] = Exporter{OTLP: otlpExporterConfig}
-
-	pipelineID := fmt.Sprintf("metrics/%s", pipeline.Name)
-	cfg.Service.Pipelines[pipelineID] = makePipelineConfig(pipeline, otlpExporterID)
+	exporterID := otlpexporter.ExporterID(pipeline.Spec.Output.Otlp, pipeline.Name)
+	cfg.Exporters[exporterID] = Exporter{OTLP: otlpExporterConfig}
 
 	return nil
 }
 
-func makePipelineConfig(pipeline *telemetryv1alpha1.MetricPipeline, exporterIDs ...string) config.Pipeline {
-	sort.Strings(exporterIDs)
-
+func makeServicePipelineConfig(pipeline *telemetryv1alpha1.MetricPipeline) config.Pipeline {
 	processors := []string{"memory_limiter", "k8sattributes"}
 
 	input := pipeline.Spec.Input
-	if !*input.Runtime.Enabled {
+	if !isRuntimeInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-runtime")
 	}
-	if !*input.Prometheus.Enabled {
+	if !isPrometheusInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-prometheus")
 	}
-	if !*input.Istio.Enabled {
+	if !isIstioInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-istio")
 	}
-	if !*input.Otlp.Enabled {
+	if input.Otlp != nil && input.Otlp.Disabled {
 		processors = append(processors, "filter/drop-if-input-source-otlp")
 	}
 
-	if shouldFilterByNamespace(input.Runtime.Enabled, input.Runtime.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceRuntime)
-		processors = append(processors, processorName)
+	if isRuntimeInputEnabled(input) && shouldFilterByNamespace(input.Runtime.Namespaces) {
+		processors = append(processors, makeNamespaceFilterID(pipeline.Name, metric.InputSourceRuntime))
 	}
-	if shouldFilterByNamespace(input.Prometheus.Enabled, input.Prometheus.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourcePrometheus)
-		processors = append(processors, processorName)
+	if isPrometheusInputEnabled(input) && shouldFilterByNamespace(input.Prometheus.Namespaces) {
+		processors = append(processors, makeNamespaceFilterID(pipeline.Name, metric.InputSourcePrometheus))
 	}
-	if shouldFilterByNamespace(input.Istio.Enabled, input.Istio.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceIstio)
-		processors = append(processors, processorName)
+	if isIstioInputEnabled(input) && shouldFilterByNamespace(input.Istio.Namespaces) {
+		processors = append(processors, makeNamespaceFilterID(pipeline.Name, metric.InputSourceIstio))
 	}
-	if shouldFilterByNamespace(input.Otlp.Enabled, input.Otlp.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceOtlp)
-		processors = append(processors, processorName)
+	if input.Otlp != nil && !input.Otlp.Disabled && shouldFilterByNamespace(input.Otlp.Namespaces) {
```

Comment:
also here we can do the check if `!isOtlpInputDisabled(input) && shouldFilterByNamespace(input.Otlp.Namespaces)`